app/controllers/SubmitVatReturnController.scala

```diff
@@ -31,7 +31,13 @@ class SubmitVatReturnController @Inject()(vatReturnsService: VatReturnsService)
                                          (implicit ec: ExecutionContext) extends BaseController {
 
   def submitVatReturn(vrn: String): Action[AnyContent] = Action.async { implicit request =>
-    val requestAsJson: Option[VatReturnDetail] = request.body.asJson.flatMap(_.asOpt[VatReturnDetail])
+    val requestAsJson: Option[VatReturnDetail] = request.body.asJson match {
+      case Some(validJson) => validJson.asOpt[VatReturnDetail]
+      case None =>
```

Comment:
Sorry to be a pain buddy, but could you a test case for this case - when the body isn't valid Json